### PR TITLE
Fix issues with broadcasting transactions - Closes #3648

### LINKED ISSUE #3648: Wrong votecast after not confirming the vote queue
### Expected behavior

It should cast votes properly.

### Actual behavior

It casts the first vote only. Breaks after that.

### Steps to reproduce

Add a delegate vote to the voting queue, click continue in the voting queue, **DON'T** click confirm. Add another delegate vote. Try to cast the whole queue. Check how many votes were really casted. After that you can't cast more votes until you relog.

### Which version(s) does this affect? (Environment, OS, etc...)

2.0.0-rc.1, Windows 10

Thanks progcppk @ discord for reporting it.

### PR DESCRIPTION
**Pull request by reyraa**
### What was the problem?
This PR resolves #3648

In the previous voting system users could vote for up to 101 delegates while each tx could only include 33 delegates. Lisk Desktop enabled users to cast all 101 votes in a single click, broadcasting 4 transactions to perform such an action.

Additionally we had a bug in Lisk Elements which caused A. race condition, B. system time mismatch. Now with the new vote system and the new version of Lisk Elements, all above issues are resolved. Therefore we don't need arrays for storing signed transaction and corresponding errors.

Given the above, there are only two situation where the retry button is useful:
 - No internet connection.
 - HW connectivity issue.

 Therefore I reduced some complexity regarding the broadcast error.

### How was it solved?
 - [x] Cleaned up the `transactions` reducer by removing all outdated action types.
 - [x] Convert tx error arrays to objects and improve names.
 - [x] Update the usages of the above error objects.
 - [x] Resolve todos
 - [x] Update unit tests

### How was it tested?
 - [x] Manually for all tx types using an ordinary account.
 - [x] Manually for all tx types using a multisignature account.
 - [x] Manually reclaim the legacy balance of an account. (The app remains in pending state because the block data structure is different on https://alphanet-v5-legacy-service.liskdev.net. This is expected).
 - [x] Unit tests.

### CODE REVIEW

**Review by isalga — APPROVED** [review]

File: src/store/reducers/transactions.js [hunk 1]
```diff
@@ -60,57 +57,52 @@ const transactions = (state = initialState, action) => { // eslint-disable-line
...
+        signedTransaction: action.data,
       };
-    // TODO can be remove after move send (create) tx to utils file
-    // istanbul ignore next
-    case actionTypes.transactionCreatedError: {
-      const { message = 'The transaction failed', name = 'TransactionFailedError' } = action.data;
+
+    // Stores the transaction signature error. This error is thrown at the time of
+    // creating the raw transaction object or signing it using Lisk Element / Bitcoin lib.
+    case actionTypes.transactionCreatedError:
```
Comment (UsamaHameed) [inline src/store/reducers/transactions.js:70]: `transactionSignError` is a better name in my opinion

File: src/store/reducers/transactions.js [hunk 2]
```diff
@@ -60,57 +57,52 @@ const transactions = (state = initialState, action) => { // eslint-disable-line
...
       return {
         ...state,
-        transactionsCreatedFailed: [...state.transactionsCreatedFailed, { message, name }],
-      }; }
-    // TODO can be remove after use HOC for send (broadcast) tx
-    // istanbul ignore next
+        txSignatureError: {
```
Comment (UsamaHameed) [inline src/store/reducers/transactions.js:73]: how is this different from `transactionCreatedError`?

**Review by UsamaHameed — CHANGES_REQUESTED** [review]
This PR required comprehensive testing, may be we can ask @sridharmeganathan for some help

File: src/components/screens/multiSignature/result/result.js [hunk 3]
```diff
@@ -2,34 +2,29 @@ import React, { useState } from 'react';
...
-    downloadJSON(transaction, `tx-${transaction.id}`);
+    downloadJSON(transaction, `tx-${transactions.signedTransaction.id}`);
   };
 
   const onCopy = () => {
     copyToClipboard(transactionToJSON(transaction));
     setCopied(true);
   };
 
+  const status = getBroadcastStatus(transactions, false); // fix the second param
```
[the remark this answers is not in the document]
Comment (reyraa) [reply src/components/screens/multiSignature/result/result.js:25]: Yes, for 2.1.0 where we handle HW integration.

[the remark this answers is not in the document]
Comment (reyraa) [reply src/components/screens/multiSignature/result/result.js:25]: I'll create a ticket now so we don't forget it.

File: src/components/screens/lockedBalance/transactionStatus/transactionStatus.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import { PrimaryButton } from '@toolbox/buttons';
+import { TransactionResult, getBroadcastStatus } from '@shared/transactionResult';
+import statusMessages from './statusMessages';
+import styles from './status.css';
+
+const TransactionStatus = (props) => {
+  const {
+    t, history, transactionInfo, transactions, transactionBroadcasted,
+  } = props;
+  const status = getBroadcastStatus(transactions, false); // handle HW error
```
[the remark this answers is not in the document]
Comment (reyraa) [reply src/components/screens/lockedBalance/transactionStatus/transactionStatus.js:12]: Good point

File: src/components/screens/multiSignature/result/result.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (reyraa) [reply src/components/screens/multiSignature/result/result.js]: #3648

File: src/components/screens/lockedBalance/transactionStatus/transactionStatus.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (reyraa) [reply src/components/screens/lockedBalance/transactionStatus/transactionStatus.js]: #3648

File: src/components/screens/reclaimBalance/modal/status/status.js [hunk 5]
```diff
@@ -38,79 +72,40 @@ const Status = ({
...
-    : {
-      title: t('Transaction failed'),
-      message: t('There was an error in the transaction. Please try again.'),
-      button: {
-        onClick: onRetry,
-        title: t('Try again'),
-        className: 'on-retry',
-      },
-    };
+  const status = getBroadcastStatus(transactions, false); // fix the second parameter
```
[the remark this answers is not in the document]
Comment (reyraa) [reply src/components/screens/reclaimBalance/modal/status/status.js]: #3648

File: src/components/screens/reclaimBalance/modal/status/status.js [hunk 6]
```diff
@@ -1,27 +1,61 @@
...
-  }
+const SuccessAction = ({
+  template, isMigrated, balance, t,
+}) => (
+  <>
+    <ul className={styles.successList}>
+      <li>
+        <span>
+          <LiskAmount
+            val={parseInt(balance, 10)}
```
[the remark this answers is not in the document]
Comment (reyraa) [reply src/components/screens/reclaimBalance/modal/status/status.js:18]: Can't commit yours, seems the diff is outdated.

File: src/store/reducers/transactions.js (hunk 1 above)
Comment (reyraa) [reply src/store/reducers/transactions.js]: Renamed.

**Review by UsamaHameed — APPROVED** [review]

**Review by sridharmeganathan — APPROVED** [review]

**Pull request merged**